samples/openthread/ncp/prj.conf

```diff
@@ -0,0 +1,146 @@
+CONFIG_BOOT_BANNER=n
+CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE=2048
+CONFIG_LOG_BACKEND_SPINEL_LEVEL_INFO=y
+# ===============================
+
+# ===============================
+# NETWORKING OPTIONS
+# ===============================
+# ====================
+# OPENTHREAD OPTIONS
+# ====================
+# set OpenThread NCP architecture
+CONFIG_OPENTHREAD_NCP=y
+
+# enable OpenThread
+CONFIG_NET_L2_OPENTHREAD=y
+
+CONFIG_OPENTHREAD_DEBUG=y
+CONFIG_OPENTHREAD_L2_DEBUG=y
+CONFIG_OPENTHREAD_L2_LOG_LEVEL_INF=y
+CONFIG_OPENTHREAD_CHANNEL=11
+# Use PANID 0xffff to be able to commission
+CONFIG_OPENTHREAD_PANID=65535
+
+# stack sizes configuration
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=5120
+CONFIG_SHELL_STACK_SIZE=3072
+CONFIG_OPENTHREAD_THREAD_STACK_SIZE=8192
+CONFIG_OPENTHREAD_PKT_LIST_SIZE=20
+
+# Other OpenThread features
+CONFIG_OPENTHREAD_DHCP6_SERVER=y
+CONFIG_OPENTHREAD_COMMISSIONER=y
+CONFIG_OPENTHREAD_BORDER_AGENT=y
+CONFIG_OPENTHREAD_BORDER_ROUTER=y
+CONFIG_OPENTHREAD_UDP_FORWARD=y
+CONFIG_OPENTHREAD_ENABLE_SERVICE=y
+
+# ====================
+
+# ====================
+# COMMON OPTIONS
+# ====================
+# Generic
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_CONFIG_NEED_IPV6=y
+
+# disable certain parts of Zephyr IPv6 stack
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+
+CONFIG_NET_TCP=n
+CONFIG_NET_IPV4=n
+CONFIG_NET_CONFIG_NEED_IPV4=n
+
+# make sure Bluetooth stack is disabled
+CONFIG_BT=n
+
+# reduce very wide OpenThread addressing register
+CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=6
+CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=8
+CONFIG_NET_MAX_CONTEXTS=10
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=16
+CONFIG_NET_PKT_TX_COUNT=16
+CONFIG_NET_BUF_RX_COUNT=80
+CONFIG_NET_BUF_TX_COUNT=80
+CONFIG_NET_CONTEXT_NET_PKT_POOL=y
+# ====================
+# ===============================
+
+# ===============================
+# INTERFACES OPTIONS
+# ===============================
+# ====================
+# USB CONFIGURATION
+# ====================
+# Use UART_1 for NCP
+CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_DEV_NAME="UART_1"
+
+# Use USB-CDC-ACM for NCP
+#CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_ACM=y
+#CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_DEV_NAME="CDC_ACM_0"
+#CONFIG_USB=y
+#CONFIG_USB_DEVICE_STACK=y
+#CONFIG_USB_CDC_ACM=y
+#CONFIG_UART_INTERRUPT_DRIVEN=y
+#CONFIG_UART_LINE_CTRL=y
+#CONFIG_USB_REQUEST_BUFFER_SIZE=2048
+
+# ===============================
+# OTHER OPTIONS
+# ===============================
+# OpenThread stack needs C++ support
+CONFIG_CPLUSPLUS=y
+
+# enable NCP to reboot device
+CONFIG_REBOOT=y
+
+# turn off OpenThread shell to not have conflict with Zephyr shell
+CONFIG_OPENTHREAD_SHELL=n
+
+# Other OpenThread dependencies
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+CONFIG_NVS=y
+CONFIG_SETTINGS=y
+
+# mbedTLS settings
+CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN=768
+CONFIG_MBEDTLS_HEAP_SIZE=8192
```

Comment:
this is too small 10K is required. After zephyr upmerge this can be removed as well proper heap size will be selected automatically

Reply:
Done